snap/snapcraft.yaml

```diff
@@ -11,42 +11,52 @@ grade: devel
 license: GPL-3.0
 source-code: https://github.com/canonical/imagecraft
 
+environment:
+  PYTHONPATH: "$SNAP/lib/python3.12/site-packages:${PYTHONPATH}"
+
 apps:
   imagecraft:
-    command: bin/python3 $SNAP/bin/imagecraft
+    command: bin/python $SNAP/bin/imagecraft
     environment:
       PATH: "/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:$SNAP/usr/bin:$SNAP/bin"
+      PYTHONHOME: "/usr/"
 
 build-packages:
   - git
   - libapt-pkg-dev
   - libyaml-dev
-  - python3.10-dev
+  - python3.12-dev
+  - python3.12-venv
   - pkg-config
+  - python3-pip
+  - python3-setuptools
+  - python3-wheel
+  - python3-venv
+  - python3-minimal
+  - python3-pkg-resources
+  - python3.12-minimal
 
 parts:
   imagecraft-libs:
     plugin: nil
     build-attributes:
       - enable-patchelf
     stage-snaps:
-      - ubuntu-image
+      - ubuntu-image/latest/edge
```

Comment:
I know we're still in testing, but is using edge here always a good idea? What's the concept here? Is it because the stable ubuntu-image isn't core24-base yet, so temporary?

Reply:
Sorry I will add a comment. This is "needed" because otherwise installing the current stable version of ubuntu-image pulls core22 and thus python3.10. And according to the craft team this might be the cause of the issue I had (python3.10 was picked up when building the `imagecraft` part instead of the theoretically default python3.12). I am really not sure about that so as soon as we land a core24 based ubuntu-image in latest/stable I will remove that because I do not want imagecraft to rely on the edge channel of ubuntu-image.